framework/src/geomsearch/NearestNodeLocator.C

```diff
@@ -212,6 +233,100 @@ NearestNodeLocator::nearestNode(dof_id_type node_id)
   return _nearest_node_info[node_id]._nearest_node;
 }
 
+void
+NearestNodeLocator::updatePatch(std::vector<dof_id_type> & recheck_slave_nodes)
+{
+  Moose::perf_log.push("NearestNodeLocator::updatePatch()", "Execution");
+
+  std::vector<dof_id_type> trial_master_nodes;
+
+  // Build a bounding box.  No reason to consider nodes outside of our inflated BB
+  MeshTools::BoundingBox * my_inflated_box = NULL;
+
+  const std::vector<Real> & inflation = _mesh.getGhostedBoundaryInflation();
+
+  // This means there was a user specified inflation... so we can build a BB
+  if (inflation.size() > 0)
+  {
+    MeshTools::BoundingBox my_box = MeshTools::create_local_bounding_box(_mesh);
+
+    Real distance_x = 0;
+    Real distance_y = 0;
+    Real distance_z = 0;
+
+    distance_x = inflation[0];
+
+    if (inflation.size() > 1)
+      distance_y = inflation[1];
+
+    if (inflation.size() > 2)
+      distance_z = inflation[2];
+
+    my_inflated_box = new MeshTools::BoundingBox(Point(my_box.first(0) - distance_x,
```

Comment:
Use a std::unique ptr here:
```
my_inflated_box = libmesh_make_unique<MeshTools::BoundingBox>(point1, point2);
```

Then remove the corresponding delete below. You never know when this will be refactored and you miss hitting the delete statement.

Reply:
Thanks for the comment Cody. I have made this change.